### DISCUSSION TITLE
LineEdit / TextEdit - no cursor in focused element with right horizontal alignment

### DISCUSSION THREAD

**Discussion by HitecExports** [slint-ui/slint#3996] [Q&A] [answered]
Hello!

Using Slint 1.3.0, Windows 11, wininit, femtovg / skia

LineEdit / TextEdit does not show cursor in focused element with **right horizontal alignment** when the text is empty, or cursor should be in rightmost position.

I use **right horizontal alignment** for entering numbers in a LineEdit.

This problem occurs in femtovg an skia as well.

![image](https://github.com/slint-ui/slint/assets/19405195/e3e894e0-39a2-41b5-baaf-1925b4348589)

I did workaround:
1. Created a copy of LineEdit as a custom Widget.
2. Changed width in the i-text-input (subtracted 1px) - to allow cursor to be shown
3. Probably the subtraction should be conditional - only for right alignment.
```
...
// Changed width in the i-text-input
i-text-input := TextInput {
        property <length> computed_x;
        property <length> padding-outer: i-layout.padding-left + i-layout.padding-right;
        x: min(0px, max(parent.width - self.width, self.computed_x));
        width: max(parent.width, self.preferred-width) - 1px;                                  // subtracted 1px - to allow cursor to be shown
        height: 100%;
...
```

But it nice to have in a standard LineEdit / TextEdit widget.

**Comment by ogoffart** [comment, marked as answer]
You're right that there is a bug there.  Fixed in #4016

**Comment by HitecExports** [comment]
Thank you for the bug fix!
